### PR TITLE
BST-6117: Dependabot server-side module

### PR DESCRIPTION
**Pull request by ledo01**
Added a server-side module & rules definitions for Dependabot. The rules were imported from native-scanner.

### CODE REVIEW

File: server-side-scanners/boostsecurityio/dependabot/rules.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  cve-moderate:
+    categories:
+    - ALL
+    - supply-chain
+    - boost-hardened
+    - vulnerable-and-outdated-components
+    description: Dependency with a Medium Risk Vulnerability
+    group: top10-vulnerable-components
+    name: cve-moderate
+    pretty_name: Dependency with a Moderate Risk Vulnerability
```
Comment (lindycoder) [inline server-side-scanners/boostsecurityio/dependabot/rules.yaml:46]: Just noticed the difference in description/pretty name, don't know if @boostsecurityio/security wanna do something about it

Comment (Becojo) [reply server-side-scanners/boostsecurityio/dependabot/rules.yaml:46]: I'd say Medium is more in line with the CVSS terminology.

```suggestion
    pretty_name: Dependency with a Medium Risk Vulnerability
```

Comment (Becojo) [reply server-side-scanners/boostsecurityio/dependabot/rules.yaml:46]: oh but the rule id is `moderate` and that can't easily change :|

**Review by Becojo — APPROVED** [review]
changing description/pretty_name is not a blocker

**Review by lindycoder — DISMISSED** [review]

**Review by fproulx-boostsecurity — APPROVED** [review]

**Pull request merged**